### PR TITLE
Use boost::test in test_equil

### PR DESCRIPTION
**Pull request by akva2**
Unify with other tests.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
jenkins build this please

File: tests/test_equil.cc [hunk 1]
```diff
@@ -450,21 +458,21 @@ void test_DeckAllDead()
 
     Opm::EQUIL::DeckDependent::InitialStateComputer<TypeTag> comp(*simulator->problem().materialLawManager(), eclipseState, simulator->vanguard().grid(), 10.0);
     const auto& pressures = comp.press();
-    REQUIRE(pressures.size() == 3);
-    REQUIRE(int(pressures[0].size()) == grid.number_of_cells);
+    BOOST_CHECK(pressures.size() == 3);
+    BOOST_CHECK(int(pressures[0].size()) == grid.number_of_cells);
```
Comment (bska) [inline tests/test_equil.cc:462]: These should really be `BOOST_REQUIRE_EQUAL` instead.  The rest of the test is meaningless unless those conditions are satisfied.

File: tests/test_equil.cc [hunk 2]
```diff
@@ -476,16 +484,16 @@ void test_CapillaryInversion()
...
     const int cell = 0;
-    const double reltol = 1.0e-7;
+    const double reltol = 1.0e-5;
     {
         const int phase = FluidSystem::waterPhaseIdx;
         const bool increasing = false;
         const std::vector<double> pc = { 10.0e5, 0.5e5, 0.4e5, 0.3e5, 0.2e5, 0.1e5, 0.099e5, 0.0e5, -10.0e5 };
         const std::vector<double> s = { 0.2, 0.2, 0.2, 0.466666666666, 0.733333333333, 1.0, 1.0, 1.0, 1.0 };
-        REQUIRE(pc.size() == s.size());
+        BOOST_CHECK(pc.size() == s.size());
```
Comment (bska) [inline tests/test_equil.cc:493]: Should be `BOOST_REQUIRE_EQUAL`.

File: tests/test_equil.cc [hunk 3]
```diff
@@ -495,10 +503,10 @@ void test_CapillaryInversion()
         const bool increasing = true;
         const std::vector<double> pc = { 10.0e5, 0.6e5, 0.5e5, 0.4e5, 0.3e5, 0.2e5, 0.1e5, 0.0e5, -10.0e5 };
         const std::vector<double> s = { 0.8, 0.8, 0.8, 0.533333333333, 0.266666666666, 0.0, 0.0, 0.0, 0.0 };
-        REQUIRE(pc.size() == s.size());
+        BOOST_CHECK(pc.size() == s.size());
```
Comment (bska) [inline tests/test_equil.cc:506]: Should be `BOOST_REQUIRE_EQUAL`.

File: tests/test_equil.cc [hunk 4]
```diff
@@ -508,15 +516,15 @@ void test_CapillaryInversion()
         const int gas = FluidSystem::gasPhaseIdx;
         const std::vector<double> pc = { 0.9e5, 0.8e5, 0.6e5, 0.4e5, 0.3e5 };
         const std::vector<double> s = { 0.2, 0.333333333333, 0.6, 0.866666666666, 1.0 };
-        REQUIRE(pc.size() == s.size());
+        BOOST_CHECK(pc.size() == s.size());
```
Comment (bska) [inline tests/test_equil.cc:519]: Should be BOOST_REQUIRE_EQUAL`.

File: tests/test_equil.cc [hunk 5]
```diff
@@ -528,33 +536,33 @@ void test_DeckWithCapillary()
     Opm::EQUIL::DeckDependent::InitialStateComputer<TypeTag> comp(*simulator->problem().materialLawManager(), eclipseState, simulator->vanguard().grid(), 10.0);
 
     const auto& pressures = comp.press();
-    REQUIRE(pressures.size() == 3);
-    REQUIRE(int(pressures[0].size()) == grid.number_of_cells);
+    BOOST_CHECK(pressures.size() == 3);
+    BOOST_CHECK(int(pressures[0].size()) == grid.number_of_cells);
```
Comment (bska) [inline tests/test_equil.cc:540]: Should be `BOOST_REQUIRE_EQUAL`.

File: tests/test_equil.cc [hunk 6]
```diff
@@ -565,24 +573,24 @@ void test_DeckWithCapillaryOverlap()
 
     Opm::EQUIL::DeckDependent::InitialStateComputer<TypeTag> comp(*simulator->problem().materialLawManager(), eclipseState, simulator->vanguard().grid(), 9.80665);
     const auto& pressures = comp.press();
-    REQUIRE(pressures.size() == 3);
-    REQUIRE(int(pressures[0].size()) == grid.number_of_cells);
+    BOOST_CHECK(pressures.size() == 3);
+    BOOST_CHECK(int(pressures[0].size()) == grid.number_of_cells);
```
Comment (bska) [inline tests/test_equil.cc:577]: Should be `BOOST_REQUIRE_EQUAL`.

File: tests/test_equil.cc [hunk 7]
```diff
@@ -603,16 +611,16 @@ void test_DeckWithCapillaryOverlap()
     s_opm[FluidSystem::oilPhaseIdx] = { 0,   0,   0,   0,   0,   0,   0,   0,                   0,                   0.20023624994125844,   0.084602875330224592, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
     s_opm[FluidSystem::gasPhaseIdx] = { 0.8, 0.8, 0.8, 0.8, 0.8, 0.8, 0.8, 0.77107068773113863, 0.46593542169947511, 0.015192997516294321, 0,      0, 0, 0, 0, 0, 0, 0, 0, 0 };
     for (int phase = 0; phase < 3; ++phase) {
-        REQUIRE(sats[phase].size() == s_opm[phase].size());
+        BOOST_CHECK(sats[phase].size() == s_opm[phase].size());
```
Comment (bska) [inline tests/test_equil.cc:614]: Should be `BOOST_REQUIRE_EQUAL`.

File: tests/test_equil.cc [hunk 8]
```diff
@@ -624,25 +632,25 @@ void test_DeckWithLiveOil()
     // Initialize the fluid system
     Opm::EQUIL::DeckDependent::InitialStateComputer<TypeTag> comp(*simulator->problem().materialLawManager(), eclipseState, simulator->vanguard().grid(), 9.80665);
     const auto& pressures = comp.press();
-    REQUIRE(pressures.size() == 3);
-    REQUIRE(int(pressures[0].size()) == grid.number_of_cells);
+    BOOST_CHECK(pressures.size() == 3);
+    BOOST_CHECK(int(pressures[0].size()) == grid.number_of_cells);
```
Comment (bska) [inline tests/test_equil.cc:636]: Should be `BOOST_REQUIRE_EQUAL`.

File: tests/test_equil.cc [hunk 9]
```diff
@@ -661,11 +669,11 @@ void test_DeckWithLiveOil()
     s_opm[FluidSystem::oilPhaseIdx] = { 0,   0,   0,   0,   0,   0,   0,   0,            0,            0.20057438297017782,   0.084716756377890667, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
     s_opm[FluidSystem::gasPhaseIdx] = { 0.8, 0.8, 0.8, 0.8, 0.8, 0.8, 0.8, 0.77083036553538653, 0.46569509476225479, 0.014706750907401245,  0,       0, 0, 0, 0, 0, 0, 0, 0, 0 };
     for (int phase = 0; phase < 3; ++phase) {
-        REQUIRE(sats[phase].size() == s_opm[phase].size());
+        BOOST_CHECK(sats[phase].size() == s_opm[phase].size());
```
Comment (bska) [inline tests/test_equil.cc:672]: Should be `BOOST_REQUIRE_EQUAL`.

File: tests/test_equil.cc [hunk 10]
```diff
@@ -699,25 +707,25 @@ void test_DeckWithLiveGas()
 
     Opm::EQUIL::DeckDependent::InitialStateComputer<TypeTag> comp(*simulator->problem().materialLawManager(), eclipseState, simulator->vanguard().grid(), 9.80665);
     const auto& pressures = comp.press();
-    REQUIRE(pressures.size() == 3);
-    REQUIRE(int(pressures[0].size()) == grid.number_of_cells);
+    BOOST_CHECK(pressures.size() == 3);
+    BOOST_CHECK(int(pressures[0].size()) == grid.number_of_cells);
```
Comment (bska) [inline tests/test_equil.cc:711]: Should be `BOOST_REQUIRE_EQUAL`.

File: tests/test_equil.cc [hunk 11]
```diff
@@ -736,11 +744,11 @@ void test_DeckWithLiveGas()
     s_opm[FluidSystem::oilPhaseIdx] =   { 0,   0,   0,   0,   0,   0,   0,   0,          0,      0.18288667, 0.0846,  0, 0, 0, 0, 0, 0, 0, 0, 0 };
     s_opm[FluidSystem::gasPhaseIdx] =   { 0.8, 0.8, 0.8, 0.8, 0.8, 0.8, 0.8, 0.75689455, 0.4612, 0.03253333, 0,       0, 0, 0, 0, 0, 0, 0, 0, 0 };
     for (int phase = 0; phase < 3; ++phase) {
-        REQUIRE(sats[phase].size() == s_opm[phase].size());
+        BOOST_CHECK(sats[phase].size() == s_opm[phase].size());
```
Comment (bska) [inline tests/test_equil.cc:747]: Should be `BOOST_REQUIRE_EQUAL`.

File: tests/test_equil.cc [hunk 12]
```diff
@@ -777,25 +785,25 @@ void test_DeckWithRSVDAndRVVD()
 
     Opm::EQUIL::DeckDependent::InitialStateComputer<TypeTag> comp(*simulator->problem().materialLawManager(), eclipseState, simulator->vanguard().grid(), 9.80665);
     const auto& pressures = comp.press();
-    REQUIRE(pressures.size() == 3);
-    REQUIRE(int(pressures[0].size()) == grid.number_of_cells);
+    BOOST_CHECK(pressures.size() == 3);
+    BOOST_CHECK(int(pressures[0].size()) == grid.number_of_cells);
```
Comment (bska) [inline tests/test_equil.cc:789]: Should be `BOOST_REQUIRE_EQUAL`.

File: tests/test_equil.cc [hunk 13]
```diff
@@ -816,11 +824,11 @@ void test_DeckWithRSVDAndRVVD()
     s_opm[FluidSystem::gasPhaseIdx]   = { 0.8, 0.8, 0.8, 0.8, 0.8, 0.8, 0.8, 0.7776954288307103, 0.47117701424054126, 0.02210249613021811,    0,          0, 0, 0, 0, 0, 0, 0, 0, 0 };
 
     for (int phase = 0; phase < 3; ++phase) {
-        REQUIRE(sats[phase].size() == s_opm[phase].size());
+        BOOST_CHECK(sats[phase].size() == s_opm[phase].size());
```
Comment (bska) [inline tests/test_equil.cc:827]: Should be `BOOST_REQUIRE_EQUAL`.

File: tests/test_equil.cc [hunk 14]
```diff
@@ -875,19 +883,19 @@ void test_DeckWithPBVDAndPDVD()
 
     Opm::EQUIL::DeckDependent::InitialStateComputer<TypeTag> comp(*simulator->problem().materialLawManager(), eclipseState, simulator->vanguard().grid(), 9.80665);
     const auto& pressures = comp.press();
-    REQUIRE(pressures.size() == 3);
-    REQUIRE(int(pressures[0].size()) == grid.number_of_cells);
+    BOOST_CHECK(pressures.size() == 3);
+    BOOST_CHECK(int(pressures[0].size()) == grid.number_of_cells);
```
Comment (bska) [inline tests/test_equil.cc:887]: Should be `BOOST_REQUIRE_EQUAL`.

File: tests/test_equil.cc [hunk 15]
```diff
@@ -904,10 +912,10 @@ void test_DeckWithPBVDAndPDVD()
     s_opm[FluidSystem::gasPhaseIdx] =   { 0.8, 0.8, 0.8, 0.8, 0.8, 0.8, 0.8, 0.75742662687407303, 0.46165175235637212, 0.033640411881804465,    0,          0, 0, 0, 0, 0, 0, 0, 0, 0 };
 
     for (int phase = 0; phase < 3; ++phase) {
-        REQUIRE(sats[phase].size() == s_opm[phase].size());
+        BOOST_CHECK(sats[phase].size() == s_opm[phase].size());
```
Comment (bska) [inline tests/test_equil.cc:915]: Should be `BOOST_REQUIRE_EQUAL`.

**Review by bska — CHANGES_REQUESTED** [review]
Very good.  I'm happy to have this unit test in a more formalised framework.  Once you address the semantic changes (`REQUIRE` changed to `BOOST_CHECK` when it should be `BOOST_REQUIRE_EQUAL` instead) I'll merge this.

**Comment by akva2** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks a lot for the updates.  This is the exact version I hoped to see.  I'll merge once the build check completes.

**Comment by bska** [comment]
> I'll merge once the build check completes.

Build check was successful.  Merging as advertised.

**Pull request merged**